tools/api-metadata-extractor/go.mod

```diff
@@ -0,0 +1,7 @@
+module api-metadata-extractor
+
+go 1.22.5
+
+toolchain go1.23.9
+
+require gopkg.in/yaml.v3 v3.0.1
```

Comment:
nit: in a lot of areas of the code base we use `sigs.k8s.io/yaml` would it makes sense for us to align on a single yaml package?